compiler/rustc_serialize/src/leb128.rs

```diff
@@ -51,28 +51,35 @@ macro_rules! impl_read_unsigned_leb128 {
     ($fn_name:ident, $int_ty:ty) => {
         #[inline]
         pub fn $fn_name(slice: &[u8], position: &mut usize) -> $int_ty {
-            // The first iteration of this loop is unpeeled. This is a
-            // performance win because this code is hot and integer values less
-            // than 128 are very common, typically occurring 50-80% or more of
-            // the time, even for u64 and u128.
-            let byte = slice[*position];
-            *position += 1;
-            if (byte & 0x80) == 0 {
-                return byte as $int_ty;
-            }
-            let mut result = (byte & 0x7F) as $int_ty;
-            let mut shift = 7;
-            loop {
+            #[inline]
+            fn inner(slice: &[u8], position: &mut usize) -> $int_ty {
```

Comment:
Why is this inner function useful?

Reply:
For the commented-out code at the bottom of it :joy: 

It's annoying to log what this code is doing as-is because it has two return paths. And this approach blows up compile times by about 20x so I'm ripping this out and maybe we'll find a better way to monitor this in another PR.